src/pages/index.js

```diff
@@ -54,77 +57,245 @@ const cardSection = new Section(
   ".cards__list"
 );
 
-cardSection.renderItems();
+// cardSection.renderItems();
 
-const addCardPopup /* */ = new PopupWithForm(
+const addCardPopup = new PopupWithForm(
   "#add-card-modal",
   handleAddCardFormSubmit
 );
-addCardPopup.setEventListeners();
-
+const avatarPopup = new PopupWithForm("#avatar-edit-modal", handleAvatarSubmit);
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
+
+addCardPopup.setEventListeners();
 editProfilePopup.setEventListeners();
+avatarPopup.setEventListeners();
 
 const imagePreviewPopup = new PopupWithImage("#preview-image-modal"); //should be good
 imagePreviewPopup.setEventListeners();
 
-const userInfor = new UserInfo({
+const userInfoInstance = new UserInfo({
   profileName: ".profile__title",
   jobElement: ".profile__description",
+  avatarImage: ".profile__image",
 });
 
-//----------------------------------------------------------------------------------------
+// Instantiating  PopupConfirmDelete project 9
+
+const confirmDelete = new PopupConfirmDelete("#delete-image-modal");
+
+confirmDelete.setEventListeners();
+
+//Project 9 /////////////////////////////////////////////////////////////////////
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c801e776-9008-430b-a7ca-bcf7d2aaaf7f", // my special token
+    "Content-Type": "application/json",
+  },
+});
+
+// api
+//   .getInitialCards()
+//   .then((data) => {
+//     cardSection.renderItems(data);
+//     console.log(data);
+//     // console.log("Fetched Cards:", data);
+//   })
+//   .catch((err) => {
+//     console.error(err);
+//   });
+
+api
+  .getUserInfoAndCards()
+  .then(({ userInfo, cards }) => {
+    userInfoInstance.setUserInfo({
+      name: userInfo.name,
+      description: userInfo.about,
+    });
+    userInfoInstance.updateAvatarImage({ avatar: userInfo.avatar });
+    cardSection.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error("Failed to load user information or cards:", err);
+    alert("Unable to load user data or cards. Please try again later.");
+  });
+
+// // Open avatar edit popup
+const avatarEditButton = document.querySelector(".profile__edit-avatar-button");
+avatarEditButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+// // Function to handle avatar update
+function handleAvatarSubmit(inputData, saveButton) {
+  renderSaving(true, saveButton);
+  api
+    .updateUserAvatar({ avatar: inputData.avatar })
+    .then((userData) => {
+      userInfoInstance.updateAvatarImage({ avatar: userData.avatar });
+      avatarPopup.close();
+      avatarFormValidator.resetValidation();
+      avatarFormElement.reset();
```

Comment:
* Please, disable the `avatar` save button after submisson as well  https://snipboard.io/IxbLls.jpg 



```js

      avatarFormElement.reset();  // reset first
      avatarFormValidator.resetValidation();  // then it will disable the button
```